src/sqlfluff/dialects/dialect_redshift.py

```diff
@@ -1120,6 +1120,161 @@ class CreateSchemaStatementSegment(BaseSegment):
     )
 
 
+@redshift_dialect.segment()
+class AltereDatashareStatementSegment(BaseSegment):
+    """A `ALTER DATASHARE` statement.
+
+    As specified in https://docs.aws.amazon.com/redshift/latest/dg/r_ALTER_DATASHARE.html
+    """
+
+    type = "create_datashare_statement"
+    match_grammar = Sequence(
+        "ALTER",
+        "DATASHARE",
+        Ref("ObjectReferenceSegment"),
+        OneOf(
+            # add or remove objects to the datashare
+            Sequence(
+                OneOf(
+                    "ADD",
+                    "REMOVE",
+                ),
+                OneOf(
+                    Sequence(
+                        "TABLE",
+                        Delimited(Ref("TableReferenceSegment")),
+                    ),
+                    Sequence(
+                        "SCHEMA",
+                        Delimited(Ref("SchemaReferenceSegment")),
+                    ),
+                    Sequence(
+                        "FUNCTION",
+                        Delimited(Ref("FunctionNameSegment")),
+                    ),
+                    Sequence(
+                        "ALL",
+                        OneOf("TABLES", "FUNCTIONS"),
+                        "IN",
+                        "SCHEMA",
+                        Delimited(Ref("SchemaReferenceSegment")),
+                    ),
+                ),
+            ),
+            # configure the properties of the datashare
+            Sequence(
+                "SET",
+                OneOf(
+                    Sequence(
+                        "PUBLICACCESSIBLE",
+                        Ref("EqualsSegment", optional=True),
+                        OneOf(
+                            "TRUE",
+                            "FALSE",
+                        ),
+                    ),
+                    Sequence(
+                        "INCLUDENEW",
+                        Ref("EqualsSegment", optional=True),
+                        OneOf(
+                            "TRUE",
+                            "FALSE",
+                        ),
+                        "FOR",
+                        "SCHEMA",
+                        Ref("SchemaReferenceSegment"),
+                    ),
+                ),
+            ),
+        ),
+    )
+
+
+@redshift_dialect.segment()
+class CreateDatashareStatementSegment(BaseSegment):
+    """A `CREATE DATASHARE` statement.
+
+    As specified in https://docs.aws.amazon.com/redshift/latest/dg/r_CREATE_DATASHARE.html
+    """
+
+    type = "create_datashare_statement"
+    match_grammar = Sequence(
+        "CREATE",
+        "DATASHARE",
+        Ref("ObjectReferenceSegment"),
+        Sequence(
+            Ref.keyword("SET", optional=True),
+            "PUBLICACCESSIBLE",
+            Ref("EqualsSegment", optional=True),
+            OneOf(
+                "TRUE",
+                "FALSE",
+            ),
```

Comment:
There's a segment for this btw:

```suggestion
            Ref("BooleanLiteralGrammar")
```